src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -318,8 +322,15 @@ private static bool IsNullCheck(IOperation operand1, IOperation operand2, IParam
             IBlockOperation? blockStatementOpt,
             CancellationToken cancellationToken)
         {
-            // First see if we can convert a statement of the form "this.s = s" into "this.s = s ?? throw ...".
-            var documentOpt = await TryAddNullCheckToAssignmentAsync(
+            // First see if we can adopt the '!!' parameter null checking syntax.
+            var documentOpt = await TryAddNullCheckToParameterDeclarationAsync(document, parameter, cancellationToken).ConfigureAwait(false);
+            if (documentOpt != null)
+            {
+                return documentOpt;
+            }
+
+            // Then see if we can convert a statement of the form "this.s = s" into "this.s = s ?? throw ...".
+            documentOpt = await TryAddNullCheckToAssignmentAsync(
```

Comment:
Nit.  Consider:

```suggestion
            // First see if we can adopt the '!!' parameter null checking syntax.
            // Then see if we can convert a statement of the form "this.s = s" into "this.s = s ?? throw ...".
            var documentOpt =
                await TryAddNullCheckToParameterDeclarationAsync(document, parameter, cancellationToken).ConfigureAwait(false) ??
                await TryAddNullCheckToAssignmentAsync(document, parameter, blockStatementOpt, cancellationToken).ConfigureAwait(false);
```

Reply:
(up to you on this though).

Reply:
It would read more nicely this way, but I prefer the current way so that it's easy to set breakpoints on each call.